netlab/L5/tls_protocol_layer.hpp

```diff
@@ -0,0 +1,670 @@
+#pragma once
+
+#include "tls_definition.hpp"
+#include "L5.h"
+
+#include <iostream>
+#include <array>
+#include <vector>
+#include <iterator>
+#include <string>
+#include <ctime>
+
+#include <openssl/ssl.h>
+#include <openssl/err.h>
+
+namespace netlab {
+
+/************************************************************************/
+/*                         TLS Record Layer                             */
+/************************************************************************/
+
+	class TLSRecordLayer {
+	public:
+
+		ProtocolVersion protocol_version;
+		ContentType content_type;
+		uint16_t length;
+		std::vector<uint8_t> fragment;
+
+		/* Constructor */
+		TLSRecordLayer()
+			: protocol_version{ 3, 3 },
+			content_type(TLS_CONTENT_TYPE_MAX_VALUE),
+			length(0),
+			fragment{}
+		{ }
+
+		/* Method used for serializing the record layer data into a string type for sending. */
+		std::string serialize_record_layer_data() {
```

Comment:
you have a lot of implementation in the .hpp file.
i think it is batter to split definition and implantation and move implementation to tls_protocl_layer.cpp.

Reply:
Good though. I thought that this is the way to go but felt like it was still pretty sparse to do it yet. But, eventually yes (even if no more additions would be made to these classes, though I find it hard to believe)